40.md

```diff
@@ -0,0 +1,59 @@
+NIP-40
+======
+
+Expiration Timestamp
+-----------------------------------
+
+`draft` `optional` `author:0xtlt`
+
+The `expiration` tag enables users to specify a unix timestamp at which the message SHOULD be considered expired (by relays and clients) and SHOULD be deleted by relays.
+
+#### Spec
+
+```
+tag: expiration
+values:
+ - [UNIX timestamp in seconds]: required
+```
+
+#### Example
+
+```json
+{
+    "pubkey": "<pub-key>",
+    "created_at": 1000000000,
+    "kind": 1,
+    "tags": [
+      ["expiration", "1600000000"]
+    ],
+    "content": "This message will expire at the specified timestamp and be deleted by relays.\n",
+    "id": "<event-id>"
+}
+```
+
+Note: The timestamp should be in the same format as the created_at timestamp and should be interpreted as the time at which the message should be deleted by relays.
+
+Client Behavior
+---------------
+
+Clients SHOULD use the `supported_nips` field to learn if a relay supports this NIP. Clients SHOULD NOT send expiration events to relays that do not support this NIP.
+
+Clients SHOULD ignore events that have expired.
+
+Relay Behavior
+--------------
+
+Relays MAY NOT delete an expired message immediately on expiration and MAY persist them indefinitely.  
+Relays SHOULD NOT send expired events to clients, even if they are stored.  
+Relays SHOULD drop any events that are published to them if they are expired.  
+Relays SHOULD persist ephemeral events until their expiration timestamp if an ephemeral event has one, and relays MAY limit the amount of time an ephemeral event is persisted.
```

Comment:
Ephemeral events currently do not get persisted. They get relaid to interested parties and dropped. With this change, they turn into non-ephemeral events. I'm against expiration for ephemeral events. If anything I would make this explicit:

```suggestion
An expiration timestamp does not affect storage of ephemeral events.
```